docs/3_finetuner_in_action/3_1_text_to_text.md

```diff
@@ -1,3 +1,119 @@
+╰───────────────────────────────────────────────────╯
+╭───────────────────── Attributes Summary ─────────────────────╮
+│                                                              │
+│   Attribute   Data type   #Unique values   Has empty value   │
+│  ──────────────────────────────────────────────────────────  │
+│   id          ('str',)    104598           False             │
+│   tags        ('dict',)   104598           False             │
+│   text        ('str',)    104559           False             │
+│                                                              │
+╰──────────────────────────────────────────────────────────────╯
+```
+
+So we have 104598 training `Document`s. Each `Document` consists of a text field that contains the question, as well as a `finetuner_label` which indicates the label to which the question belongs. If multiple questions have the same label, they are duplicates of one another. If they have different `finetuner_label`s, they are not duplicates of each other.
+
+As for the evaluation dataset, we load `query_data` and `index_data` seperately. The `query_data` has the same structure as the `train_data`, consisting of labelled documents. The `index_data` is the data against which the queries will be matched, and contains many documents, some of which may be irrelevant to the queries (ie. they have no duplicated in the `query_data`).
+If you look at the summaries for the `query_data` and `index_data`, you will find that they have the following number of instances:
+
+```
+Length of queries DocumentArray: 5000
+Length of index DocumentArray: 15746
+```
+
+
+## Finetune the Bert model
+
+Now that we have the training and evaluation datasets loaded as `DocumentArray`s, we can start our fine-tuning experiment with this dataset using Bert.
+
+```python
+import finetuner
+from finetuner.callbacks import EvaluationCallback
+
+# Start fine-tuning as a run within an experiment
+finetuner.fit(
+    model='bert-base-cased',
+    train_data=train_data,
+    experiment_name='finetune-quora-dataset',
+    run_name='finetune-quora-dataset-distilbert-1',
+    description='this is a trial run on quora dataset with bert-base-cased.',
+    loss='TripletMarginLoss',
+    miner='TripletMarginMiner',
+    optimizer='Adam',
+    learning_rate = 1e-5,
+    epochs=3,
+    batch_size=128,
+    scheduler_step='batch',
+    freeze=False, # We are training the whole bert model, not an additional MLP.
+    output_dim=None,
+    cpu=False,
+    num_workers=4,
+    callbacks=[EvaluationCallback(query_data=query_data, index_data=index_data, batch_size=256)]
+)
+```
+
+You can check the status of your experiment and save the model when the experiment has completed.
+
+```python
+experiment = finetuner.get_experiment('finetune-quora-dataset')
+run = experiment.get_run('finetune-quora-dataset-distilbert-1')
+print(f'Run status: {run.status()}')
+print(f'Run logs: {run.logs()}')
+
+run.save_model('.')
```

Comment:
This won't work. You need to provide a path with a model name. something like "tuned_model" or "models/tuned_model"